src/atomate2/vasp/flows/eos/base.py

```diff
@@ -0,0 +1,207 @@
+    postprocessor : .job
+        optional postprocessing step
+    """
+
+    name: str = "EOS Maker"
+    relax_maker: Maker = field(default_factory=EosDoubleRelaxMaker)
+    deformation_maker: Maker = field(
+        default_factory=lambda: DeformationMaker(
+            copy_vasp_kwargs={"additional_vasp_files": ("WAVECAR",)}
+        )
+    )
+    static_maker: Maker | None = None
+    linear_strain: tuple[float, float] = (-0.05, 0.05)
+    number_of_frames: int = 6
+    postprocessor: Job = postprocess_EOS
+
+    def make(self, structure: Structure, prev_dir: str | Path | None = None):
+        """
+        Run an EOS VASP job.
+
+        Parameters
+        ----------
+        structure : Structure
+            A pymatgen structure object.
+        prev_dir : str or Path or None
+            A previous VASP calculation directory to copy output files from.
+        """
+        jobs: dict[str, list[Job]] = {"relax": [], "static": [], "postprocess": []}
+
+        relax_flow = self.relax_maker.make(structure=structure, prev_dir=prev_dir)
+        relax_flow.name = "EOS equilibrium relaxation"
+        equil_props = {
+            "relax": {
+                "E0": relax_flow.output.calcs_reversed[0].output.energy,
+                "V0": relax_flow.output.calcs_reversed[0].output.structure.volume,
+            }
+        }
+
+        if self.static_maker:
+            equil_static = self.static_maker.make(
+                structure=relax_flow.output.structure,
+                prev_dir=relax_flow.output.dir_name,
+            )
+            equil_static.name = "EOS equilibrium static"
+            equil_props["static"] = {
+                "E0": equil_static.output.calcs_reversed[0].output.energy,
+                "V0": equil_static.output.calcs_reversed[0].output.structure.volume,
+            }
+
+        strain_l, strain_delta = np.linspace(
+            self.linear_strain[0],
+            self.linear_strain[1],
+            self.number_of_frames,
+            retstep=True,
+        )
+        # Cell without applied strain already included from relax/equilibrium
+        # steps. Perturb this point (or these points) if included
+        zero_strain_mask = np.abs(strain_l) < 1.0e-15
+        if np.any(zero_strain_mask):
+            nzs = len(strain_l[zero_strain_mask])
+            shift = strain_delta / (nzs + 1.0) * np.linspace(-1.0, 1.0, nzs)
+            strain_l[np.abs(strain_l) < 1.0e-15] += shift
+
+        insert_equil_index = np.searchsorted(strain_l, 0, side="left")
+        deformation_l = [(np.identity(3) * (1 + eps)).tolist() for eps in strain_l]
+
+        for ideformation, deformation in enumerate(deformation_l):
+            # Doubly ensure that relaxations are done at fixed volume --> ISIF = 2
+            deform_relax_maker = update_user_incar_settings(
+                flow=self.deformation_maker, incar_updates={"ISIF": 2}
+            )
+            deform_relax_job = deform_relax_maker.make(
```

Comment:
Can we copy the logic elastic workflow and generate the deformations in a separate job. This will mean we won't need to implement a `DeformationMaker` for each code (VASP, CP2K, etc). You can just use a regular maker. In the elastic workflow we also write the transformation information to a file in the directory so it easy to track which deformations have been applied.

E.g., see here: https://github.com/materialsproject/atomate2/blob/8746633bf1f597deefc2dd52463bcc3272213cbe/src/atomate2/common/jobs/elastic.py#L103-L163

Reply:
I've more or less copied the elastic workflow logic here. I found during testing that using a function like `atomate2.common.jobs.elastic.run_elastic_deformations` resulted in incorrect flow logic when trying to perform statics on the deformed+relaxed structures. (The postprocessing step would happen simultaneously with the statics, despite it relying on energies/volumes from the statics.)

For that reason, the deformations are applied in an external job, `atomate2.common.jobs.eos.apply_strain_to_structure`, but the relax/static jobs are handled in the flow itself (`atomate2.common.flows.eos.CommonEosMaker`)